source/data-api/generated-endpoints.txt

```diff
@@ -0,0 +1,1364 @@
+
+Authentication
+~~~~~~~~~~~~~~
+
+Data API endpoints run in the context of a specific user, which allows
+your app to enforce rules and validate document schemas for each
+request.
+
+By default, endpoints use Application Authentication, which requires
+each request to include credentials for one of your application users,
+like an API key or JWT. You can also configure other custom
+authentication schemes to fit your application's needs.
+
+.. tabs::
+
+   .. tab:: Application
+      :tabid: auth-credentials
+
+      Application authentication requires users to authenticate each
+      Data API request by providing login credentials for one of the
+      following supported authentication providers:
+      
+      - :ref:`Email/Password <email-password-authentication>`
+      - :ref:`API Key <api-key-authentication>`
+      - :ref:`Custom JWT <custom-jwt-authentication>`
+      
+      To see how to include authentication credentials in Data API
+      requests, see :ref:`data-api-authenticate`.
+      
+      .. note::
+         
+         You must enable an authentication provider before users can
+         authenticate through it.
+   
+   .. tab:: User ID
+      :tabid: auth-userid
+
+      User ID authentication runs all Data API requests as a single,
+      pre-selected application user. To select the user, specify their
+      User Account ID in your app's Data API configuration.
+      
+      .. code-block:: json
+         :caption: http_endpoints/data_api_config.json
+
+         {
+           "run_as_user_id": "628e47baf4c2ac2796fc8a91"
+         }
+   
+   .. tab:: Script
+      :tabid: auth-script
+   
+      Script authentication calls a function to determine which
+      application user a Data API request runs as.
+
+      The function must return an existing application user's Account ID
+      as a string or ``{ "runAsSystemUser": true }`` to run the request
+      as a :ref:`system user <system-user>` that has full access to
+      MongoDB CRUD and Aggregation APIs and is not affected by any
+      rules, roles, or permissions.
+      
+      .. code-block:: json
+         :caption: http_endpoints/data_api_config.json
+
+         {
+           "run_as_user_id_script_source": "exports = () => {return \"628e47baf4c2ac2796fc8a91\"}"
+         }
+
+.. _data-api-response-type:
+
+Response Type
+~~~~~~~~~~~~~
+
+Endpoints can return data in one of two :ref:`data formats
+<data-api-data-formats>`, either JSON or EJSON.
+
+By default, endpoints return JSON, which is a standard data format that
+is widely supported modern langauges and platforms. However, JSON cannot
+represent every data type that you can store in MongoDB and loses type
+information for some data types.
+
+You can also configure endpoints to return EJSON, which uses structured
+JSON objects to fully represent the types that MongoDB supports. This
+preserves type information in responses but requires that your
+application understands how to parse and use EJSON.
```

Comment:
worth stating that the mongodb drivers all have ejson parsers and linking to this page - https://www.mongodb.com/docs/v5.3/reference/mongodb-extended-json/